torch/testing/_internal/common_methods_invocations.py

```diff
@@ -3374,6 +3374,37 @@ def sample_inputs_linalg_solve(op_info, device, dtype, requires_grad=False, vect
         out.append(SampleInput(a, args=(b,)))
     return out
 
+def sample_inputs_linalg_solve_triangular(op_info, device, dtype, requires_grad=False, **kwargs):
+    make_arg = partial(make_tensor, dtype=dtype, device=device)
+    bs = (1, 2, 0)
+    ns = (3, 0)
+    ks = (1, 3, 0)
+
+    def gen_inputs():
+        for b, n, k in itertools.chain(product(bs, ns, ks)):
+            for left, upper, uni in product((True, False), repeat=3):
+                with torch.no_grad():
+                    if b == 1:
+                        A = make_arg((n, n)) if left else make_arg((k, k))
```

Comment:
should we also test transposed inputs?

Reply:
A may be copied if out is of the correct size and, after doing the transformations on out to leave it F-contig, the resulting A is still F-contig and conjugated. Note that when out is of the correct size, resize_output is a noop.

Reply:
I kept it easy for the backward as to not make the tests run forever. As long as the forward is tested properly, the backward is just a composition of functions that are already tested.

Reply:
I second to transposed/conjugated inputs for completeness. The inputs are pretty small as to not cause that much of a slowdown.

Reply:
@Lezcano, what do you think about adding these tests? Since OpInfo testing is the preferred way of testing, let's probably update these tests to become more comprehensive.

Reply:
I run all the `OpInfo` tests locally  with all the samples and they all pass, but they are extremely slow. There is a JIT test that took 30+ min in my box.

I'd say it's fine to leave them as it is. Those test mostly test the backward and some jit stuff, and the backward is just a composition of things that are already tested with uncommon inputs.

Reply:
30+ minutes is an extremely long time for a test to run. Which test exactly was it?

Reply:
How long do the new OpInfo tests for this op take in aggregate?

Reply:
Around 1 min.